src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py

```diff
@@ -503,6 +553,198 @@ def show_release_directives(
                 )
             return cursor.fetchall()
 
+    def get_existing_app_info(self, name: str, role: str) -> dict | None:
+        """
+        Check for an existing application object by the same name as in project definition, in account.
+        It executes a 'show applications like' query and returns the result as single row, if one exists.
+        """
+        with self._use_role_optional(role):
+            try:
+                object_type_plural = ObjectType.APPLICATION.value.sf_plural_name
+                show_obj_query = f"show {object_type_plural} like {identifier_to_show_like_pattern(name)}".strip()
+
+                show_obj_cursor = self._sql_executor.execute_query(
+                    show_obj_query, cursor_class=DictCursor
+                )
+
+                show_obj_row = find_first_row(
+                    show_obj_cursor,
+                    lambda row: row[NAME_COL] == unquote_identifier(name),
```

Comment:
I think you'll have a more robust comparison if you use `_same_identifier` method for equality check here.